Feature/1017 data generator output name

dk1844 opened the pull request:
This is an update of #1047, it introduces a few changes:
 - range values are not expressed as `FROM-TO/STEP`, but `FROM-TO@STEP`. This prevented being used on FSs where files cannot contain `/`.
 - new option `-t`/`--output` is introduced in order for the user to be able to choose their own output filename in a controlled manned and would not have to guess or derive what the result will be.

### Sidenote
My ultimate aim is to merge this, later with #1047, too. This will allow having a docker image of the data-generator later usable in related project [Spline Getting Started](https://github.com/AbsaOSS/spline-getting-started/) for performance measuring.

dk1844 committed:
#1017 custom output path, range is now 1-2by3 instead of 1-2/3 to avoid generating invalid paths

dk1844 committed:
Merge branch 'feature/spline-1017-docker-generator' into feature/1017-data-generator-outputName

dk1844 committed:
#1017 numeric value is now expressed as x-y@z, reflected in Dockerfile.

dk1844 committed:
#1017 Using a predefined output file name for docker-tests

dk1844 committed:
Merge branch 'feature/spline-1017-docker-generator' into feature/1017-data-generator-outputName

dk1844 committed:
Merge branch 'feature/spline-1017-docker-generator' into feature/1017-data-generator-outputName

cerveada commented on test-data-generator/src/main/scala/za/co/absa/spline/testdatagen/GenCLI.scala:
```diff
@@ -101,9 +105,14 @@ object GenCLI {
-        s"${config.reads}reads-" +
-        s"${config.operations}ops-" +
-        s"${config.attributes}attr")
+      config.customOutputFileName.fold {
+        val defaultPrefixName = s"${config.graphType.name}-lineage-" +
+          s"${config.reads}reads-" +
+          s"${config.operations}ops-" +
+          s"${config.attributes}attr"
+        new FileDispatcher(defaultPrefixName) // using default prefix and default built-in suffix
+      } { customFileName =>
+        new FileDispatcher(customFileName, "") // custom name as-is
```
the filename creation seems to be somehow spread over two classes. I think better would be to just have a method for creating the default filename (including the suffix).
```Scala
new FileDispatcher(createDefaultFileName(config))
```

Then you don't need `fileNameSuffix` in FileDispatcher

dk1844 replied:
👍 Good idea, done.

dk1844 committed:
#1017 FileDispatcher file name creation externalized. Testable with e.g.
`java -jar .\target\test-data-generator-1.0.0-SNAPSHOT.jar -g triangle -r 5 -o 3 -a 3 -t testing_lineage.json`
`java -jar .\target\test-data-generator-1.0.0-SNAPSHOT.jar -g triangle -r 5 -o 3 -a 3`
